### PR TITLE
Implement azure vault provider

### PR DESCRIPTION
**Pull request by kroonprins**
For #3

### CODE REVIEW

File: pkg/kubeseal-convert/handlers/azurekeyvault/azurekeyvault.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+func getSecret(client *secrets.Client, secretName string) map[string]interface{} {
+	secret, err := client.GetSecret(context.TODO(), secretName, "", &secrets.GetSecretOptions{})
+	if err != nil {
+		log.Fatalf("Unable to read secret %s from the Azure Key Vault: %v", secretName, err)
+	}
+
+	mp := make(map[string]interface{})
+	error := json.Unmarshal([]byte(*secret.Value), &mp)
+	if error != nil {
+		mp[secretName] = *secret.Value
```
Comment (EladLeev) [inline pkg/kubeseal-convert/handlers/azurekeyvault/azurekeyvault.go:40]: No need to fill out the map - I used `map[string]interface{}` as I don't have a concrete struct for the returning secret.
Later on, I'm using the secret data in order to build a k8s secret and pass it to `kubeseal` command:
[buildDataBytes](https://github.com/EladLeev/kubeseal-convert/blob/main/pkg/kubeseal-convert/handlers/kubesealconvert/secret.go#L11)

This is how we end up using all the secret KV content

File: README.md [hunk 2]
```diff
@@ -54,15 +54,18 @@ The `SealedSecret` will be printed to `STDOUT`. You can run it as is, as part of
 
 ## Supported SM Systems
 :white_check_mark: AWS Secrets Manager  
-:white_check_mark: Hashicorp Vault  
+:white_check_mark: Hashicorp Vault
+:white_check_mark: Azure Key Vault
```
Comment (EladLeev) [inline README.md:58]:
```suggestion
:white_check_mark: Azure Key Vault  - contributed by @kroonprins
```

Comment (EladLeev) [reply README.md:58]: I want to praise the contributors! 🤩

**Review by EladLeev — CHANGES_REQUESTED** [review]

**Comment by EladLeev** [comment]
Thank you so much @kroonprins 🤩 
I added a few comments, can you please address them before merging?

**Comment by kroonprins** [comment]
@EladLeev , new version pushed (I will squash the commits once all comments resolved).
For the first 2 comments, I changed the implementation for retrieving the secrets. My approach was a bit wrong there I think, because the terminology in aws/hashicorp is a bit different than in azure key vault (a 'secretName' in azure key vault is just one secret value, not a collection of secrets).
I also made one small extra adaption to change the command name to 'akv' (instead of 'az') as AKV seems to be an abbreviaion they use from time to time in the microsoft docs so seems more suitable.

**Comment by EladLeev** [comment]
LGTM.
Thank you for that 🙏

**Pull request merged**